### PR TITLE
feat: nuxt-uploadthing module

### PR DESCRIPTION
**Pull request by juliusmarminge**
cc @danielroe

### CODE REVIEW

File: .npmrc [hunk 1]
```diff
@@ -11,3 +11,5 @@ public-hoist-pattern[]=*nuxt*
 public-hoist-pattern[]=defu
 public-hoist-pattern[]=ufo
 public-hoist-pattern[]=unctx
+
+shamefully-hoise=true
```
Comment (danielroe) [inline .npmrc:15]:
```suggestion
shamefully-hoist=true
```

File: .npmrc [hunk 2]
```diff
@@ -11,3 +11,5 @@ public-hoist-pattern[]=*nuxt*
 public-hoist-pattern[]=defu
 public-hoist-pattern[]=ufo
 public-hoist-pattern[]=unctx
+
+shamefully-hoist=true
```
Comment (juliusmarminge) [inline .npmrc:15]: Preferably not shamefully hoist, put all necessary packages as hoist patterns instead?

Comment (danielroe) [reply .npmrc:15]: yes, I'd agree. and I _think_ we should be able to avoid hoisting with latest nuxt...

**Comment by juliusmarminge** [comment]
@danielroe would you be willing to help me get this ready? You recommended we importing types and referece those etc and I didn't quite understand how that was supposed to happen

**Comment by juliusmarminge** [comment]
Current dependencies on/for this PR:
* main
  * **PR #293** <a href="https://app.graphite.dev/github/pr/pingdotgg/uploadthing/293" target="_blank">Graphite</a> 
    * **PR #401** <a href="https://app.graphite.dev/github/pr/pingdotgg/uploadthing/401" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/pingdotgg/uploadthing/401?utm_source=stack-comment).

**Pull request merged**